webapp/src/main/webapp/src/app/groups/groups.component.ts

```diff
@@ -12,28 +12,54 @@ import { Group } from "./model/group.model";
 export class GroupsComponent implements OnInit {
   filterOptions: GroupFilterOptions;
   groups: Group[];
+  filteredGroups: Group[];
   query: GroupQuery;
+  searchTerm: string = '';
 
-  constructor(private route: ActivatedRoute, private service: GroupService) {
+  constructor(private route: ActivatedRoute,
+              private router: Router,
+              private service: GroupService) {
   }
 
   ngOnInit() {
     this.filterOptions = this.route.snapshot.data[ "filterOptions" ];
-
     this.query = new GroupQuery(this.filterOptions.subjects);
 
-    this.query.school = this.filterOptions.schools[ 0 ];
-    this.query.schoolYear = this.filterOptions.schoolYears[ 0 ];
-    this.query.subject = this.filterOptions.subjects[ 0 ];
+    if(this.filterOptions.schools.length == 0)
+      return;
+
+    this.route.params.subscribe(p => {
```

Comment:
I'm subscribing to the route params so that whenever we change the params the following code will be called.  Otherwise on back / forward button clicks the params were not updating (since I'm using a named child route)